src/misc/scanner.ts

```diff
@@ -0,0 +1,133 @@
+import { type DetectedBarcode } from '@undecaf/barcode-detector-polyfill'
+import type { Ref } from 'vue'
+
+import { eventOn } from './callforth'
+import { DropImageFetchError } from './errors'
+import type { Point } from '../types/types'
+
+export const adaptOldFormat = (detectedCodes: DetectedBarcode[]) => {
+  if (detectedCodes.length > 0) {
+    const [firstCode] = detectedCodes
+
+    const [topLeftCorner, topRightCorner, bottomRightCorner, bottomLeftCorner] =
+      firstCode.cornerPoints
+
+    return {
+      content: firstCode.rawValue,
+      location: {
+        topLeftCorner,
+        topRightCorner,
+        bottomRightCorner,
+        bottomLeftCorner,
+
+        // not supported by native API:
+        topLeftFinderPattern: {},
+        topRightFinderPattern: {},
+        bottomLeftFinderPattern: {}
+      },
+      imageData: null
+    }
+  } else {
+    return {
+      content: null,
+      location: null,
+      imageData: null
+    }
+  }
+}
+
+/**
+ * Continuously extracts frames from camera stream and tries to read
+ * potentially pictured QR codes.
+ */
+export const keepScanning = async (
+  videoElement: Ref<HTMLVideoElement | undefined>,
+  {
+    detectHandler,
+    locateHandler,
+    minDelay
+  }: { detectHandler: any; locateHandler: any; minDelay: any }
+) => {
+  const barcodeDetector = new (await import('barcode-detector')).default({ formats: ['qr_code'] })
```

Comment:
Why do you use a dynamic import here?

Reply:
To not pollute `global` and to save on bundle size. But maybe that's necessary for `<script>` inclusion? I never added dependencies without a package manager, so could be that you know more about this? :pray:

Reply:
I would argue polyfills/shims are the exception, because their purpose is specifically to patch `global` and “pretend” that you are running in a more modern browser. That being said, this polyfill is not modifying `global` on import. So importing it the normal way is not polluting. 

Some other things to consider:

* is there a performance penalty for re-importing before every scan? The browser probably caches the asset but still.
* If we only import right before scanning there might be some awkward delay where the camera is loaded and ready but it takes some more time before the scanning actually works

Reply:
Reverting that showed that the change in https://github.com/gruhn/barcode-detector/pull/8 is necessary. I'll update this PR once the other PR is merged :)

Reply:
As @Sec-ant [pointed out](https://github.com/gruhn/barcode-detector/issues/6#issuecomment-1588414162) I was able to make use of `@sec-ant/barcode-detector` now. Let me know what you think! :)

Reply:
Eventhough I tested it locally before, using barcode-detector v1.0.2 still results in an error `unexpected token 'export'`: https://github.com/dargmuesli/vue-qrcode-reader/actions/runs/5251319091/jobs/9486148441#step:6:174 :thinking: if the previously mentioned sec-ant issue got fixed, would it be fine to move to that library?

Reply:
The mentioned issue is fixed now.